emission/analysis/modelling/trip_model/forest_classifier.py

```diff
@@ -0,0 +1,177 @@
+            'use_trip_clusters',
+        ]
+
+        for k in random_forest_expected_keys:
+            if config.get(k) is None:
+                msg = f"forest trip model config missing expected key {k}"
+                raise KeyError(msg)   
+        
+        if config['loc_feature'] == 'cluster':
+            for k in cluster_expected_keys:
+                if config.get(k) is None:
+                    msg = f"cluster trip model config missing expected key {k}"
+                    raise KeyError(msg)
+        self.model=ForestClassifierModel(config=config)
+        
+
+    def fit(self,trips: List[ecwc.Confirmedtrip]):
+        '''
+        trips : List of Entry type data 
+        '''
+        # check and raise exception if no data to fit
+        logging.debug(f'fit called with {len(trips)} trips')
+
+        unlabeled = list(filter(lambda t: len(t['data']['user_input']) == 0, trips))
+        if len(unlabeled) > 0:
+            msg = f'model.fit cannot be called with unlabeled trips, found {len(unlabeled)}'
+            raise Exception(msg)    
+        
+        #Convert List of Entry to dataframe     
+        data_df = estb.BuiltinTimeSeries.to_data_df("analysis/confirmed_trip",trips)
+        labeled_trip_df = esdtq.filter_labeled_trips(data_df)
+        expanded_labeled_trip_df= esdtq.expand_userinputs(labeled_trip_df)
+        #fit models on dataframe
+        self.model.fit(expanded_labeled_trip_df)       
+
+
+    def predict(self, trip: List[float]) -> Tuple[List[Dict], int]:
+        '''
+        trip : A single trip whose mode, pupose and replaced mode are required
+        returns.
+        '''
+
+        #check if theres no trip to predict        
+        logging.debug(f"forest classifier predict called with {len(trip)} trips")
+        if len(trip) == 0:
+            msg = f'model.predict cannot be called with an empty trips'
+            raise Exception(msg)        
+        # CONVERT LIST OF TRIPS TO dataFrame 
+        test_df = estb.BuiltinTimeSeries.to_data_df("analysis/confirmed_trip",[trip])
+        labeled_trip_df = esdtq.filter_labeled_trips(test_df)
+        expanded_labeled_trip_df= esdtq.expand_userinputs(labeled_trip_df)
+        predcitions_df= self.model.predict(expanded_labeled_trip_df)
+
+        # the predictions_df currently holds the highest probable options
+        # individually in all three categories. the predictions_df are in the form 
+        #
+        # purpose_pred | purpose_proba | mode_pred | mode_proba | replaced_pred | replaced proba 
+        # dog-park     |   1.0         |  e-bike   | 0.99       | walk          | 1.1 
+        #
+        #
+        # However, to keep the trip model general, the forest model is expected to return  
+        #
+        #PREDICTIONS [  {'labels': {'mode_confirm': 'e-bike', 'replaced_mode': 'walk', 'purpose_confirm': 'dog-park'},
+        #                 'p':  ( Currently average of the 3 probabilities)}]
+        labels= {
```

Comment:
I don't see why we have to change this to the average of the three probabilities instead of having separate ones. I don't see why this is more general than the previous approach.

Reply:
The pipeline expects the output from `predict_labels_with_n` function in `run_model.py` to be of the form ( figured this by running greedySimialrityBinning):

```
 [     {
'labels': {'mode_confirm': 'value1', 'replaced_mode': value2', 'purpose_confirm': 'value3'},

  'p':  ( a single value)}]
```

  
 Since, all three values are predicted separately they have different probs.  We can use different ways to get a single value from these three, but Avg seemed to be a good start.